metaserver/src/main/scala/scala/meta/languageserver/search/SymbolIndex.scala

```diff
@@ -49,66 +50,107 @@ class SymbolIndex(
       name <- document.names.collectFirst {
-        case name @ ResolvedName(Some(position), sym, _) if {
-              val pos = input.toIndexRange(position.start, position.end)
-              logger.info(
-                s"$sym at ${document.filename
-                  .replaceFirst(".*/", "")}:${pos.startLine}:${pos.startColumn}-${pos.endLine}:${pos.endColumn}"
+        case name @ ResolvedName(Some(position), symbol, _) if {
+              val range = input.toIndexRange(position.start, position.end)
+              logger.debug(
+                s"${document.filename.replaceFirst(".*/", "")} [${range.pretty}] ${symbol}"
               )
-              pos.startLine <= line &&
-              pos.startColumn <= column &&
-              pos.endLine >= line &&
-              pos.endColumn >= column
+              range.contains(line, column)
             } =>
           name
       }
-      msym = Symbol(name.symbol)
-      _ = logger.info(s"Found matching symbol $msym")
-      symbol <- symbols.get(name.symbol).orElse {
-        val alts = alternatives(msym)
-        logger.info(s"Trying alternatives: ${alts.mkString(" | ")}")
-        alts.collectFirst { case symbols(alternative) => alternative }
-      }
-      _ = logger.info(
-        s"Found matching symbol index ${symbol.name}: ${symbol.signature}"
-      )
-    } yield symbol
+    } yield name
   }
 
-  /** Returns the definition position of the symbol at the given position */
-  def goToDefinition(
+  /** Returns a symbol at the given location */
+  def findSymbol(
       path: AbsolutePath,
       line: Int,
       column: Int
-  ): Option[DefinitionResult] = {
+  ): Option[Symbol] = {
     for {
-      symbol <- findSymbol(path, line, column)
-      definition <- symbol.definition
-    } yield {
-      val nonJarDefinition: Position =
-        if (definition.uri.startsWith("jar:file")) {
-          definition.withUri(
-            createFileInWorkspaceTarget(URI.create(definition.uri)).toString
-          )
-        } else definition
-      logger.info(s"Found definition $nonJarDefinition")
-      val location = nonJarDefinition.toLocation
-      DefinitionResult(location :: Nil)
-    }
+      name <- resolveName(path, line, column)
+      symbol = Symbol(name.symbol)
+      _ = logger.info(s"Matching symbol ${symbol}")
+    } yield symbol
+  }
+
+  /** Returns symbol definition data from the index taking into account relevant alternatives */
+  def definitionData(
+      symbol: Symbol
+  ): Option[SymbolData] = {
+    (symbol :: alternatives.forDefinition(symbol))
+      .collectFirst {
+        case symbolIndexer(data) if data.definition.nonEmpty =>
+          logger.info(s"Found definition symbol ${data.symbol}")
+          data
+      }
+  }
+
+  /** Returns the definition location of the given symbol index data
+   * @param symbolData symbol data retrieved with [[definitionData]]
+   */
+  // TODO(alexey) this is not index-specific and should be moved to some SymbolData-ops
```

Comment:
Yep, when I said "ops" I actually meant "enrichment methods" (is it an established term?). The only doubt I had here is that `ScalametaEnrichments` is named like it's just about enrichments for the Scalameta types. Analogously to `langserver.InputEnrichments`. So should I still add it to `ScalametaEnrichments` or create a new `IndexDataEnrichments` or something?

Reply:
I think it makes sense to have only one `Enrichments` object where we keep all enrichments, that should make it easy to know "what import do I need?". We can call the object `syntax` fwiw. InputEnrichments should probably be turned into an object with a vanilla method, it's there only because it needs access to a package private method on Input. The enrichment method can be moved to the single enrichment method, wdyt?

Reply:
I like this approach 👌

Reply:
Let's move this refactoring to a separate PR

Reply:
Done in b94b0b621b450786111c3d227d2ed8a827a5889c. Actually this method disappeared.